sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -18,126 +18,95 @@
+  private long numDocs;
+  private ConnectionConfiguration connectionConfiguration;
+  private RestClient restClient;
+  private long averageDocSize;
+  private boolean insertTestDocuments;
 
-  @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
-  @Rule
-  public TestPipeline pipeline = TestPipeline.create();
+  private TestPipeline pipeline;
+  private ExpectedException expectedException;
 
-  @BeforeClass
-  public static void beforeClass() throws IOException {
-    ServerSocket serverSocket = new ServerSocket(0);
-    int esHttpPort = serverSocket.getLocalPort();
-    serverSocket.close();
-    LOG.info("Starting embedded Elasticsearch instance ({})", esHttpPort);
-    Settings.Builder settingsBuilder =
-        Settings.settingsBuilder()
-            .put("cluster.name", "beam")
-            .put("http.enabled", "true")
-            .put("node.data", "true")
-            .put("path.data", TEMPORARY_FOLDER.getRoot().getPath())
-            .put("path.home", TEMPORARY_FOLDER.getRoot().getPath())
-            .put("node.name", "beam")
-            .put("network.host", ES_IP)
-            .put("http.port", esHttpPort)
-            .put("index.store.stats_refresh_interval", 0)
-            // had problems with some jdk, embedded ES was too slow for bulk insertion,
-            // and queue of 50 was full. No pb with real ES instance (cf testWrite integration test)
-            .put("threadpool.bulk.queue_size", 100);
-    node = new Node(settingsBuilder.build());
-    LOG.info("Elasticsearch node created");
-    node.start();
-    connectionConfiguration =
-      ElasticsearchIO.ConnectionConfiguration.create(
-        new String[] {"http://" + ES_IP + ":" + esHttpPort}, ES_INDEX, ES_TYPE);
-    restClient = connectionConfiguration.createClient();
+  ElasticsearchIOTestCommon(ConnectionConfiguration connectionConfiguration, RestClient restClient,
+      long numDocs, long averageDocSize, boolean insertTestDocuments) {
+    this.connectionConfiguration = connectionConfiguration;
+    this.restClient = restClient;
+    this.numDocs = numDocs;
+    this.averageDocSize = averageDocSize;
+    this.insertTestDocuments = insertTestDocuments;
   }
 
-  @AfterClass
-  public static void afterClass() throws IOException{
-    restClient.close();
-    node.close();
+  // lazy init of the test rules (cannot be static)
+  void setPipeline(TestPipeline pipeline) {
+    this.pipeline = pipeline;
   }
 
-  @Before
-  public void before() throws Exception {
-    ElasticSearchIOTestUtils.deleteIndex(ES_INDEX, restClient);
+  void setExpectedException(ExpectedException expectedException) {
+    this.expectedException = expectedException;
   }
 
-  @Test
-  public void testSizes() throws Exception {
-    ElasticSearchIOTestUtils.insertTestDocuments(ES_INDEX, ES_TYPE, NUM_DOCS, restClient);
+  void testSizes() throws Exception {
+    if (insertTestDocuments) {
```

Comment:
It's weird that this repeats so much; perhaps the caller should explicitly call it if they want to.

Reply:
See comment above